src/main/java/com/coveo/pushapiclient/PushSource.java

```diff
@@ -0,0 +1,300 @@
+     *                  Administration Console</a>. The URL should contain your
+     *                  <code>ORGANIZATION_ID</code> and <code>SOURCE_ID</code>,
+     *                  which are required parameters for all operations regarding
+     *                  your source.
+     *                  <p>
+     *                  Some examples of valid source URLs:
+     *
+     *                  <pre>
+     * https://api.cloud.coveo.com/push/v1/organizations/my-org-if/sources/my-source-id/documents
+     * https://api-eu.cloud.coveo.com/push/v1/organizations/my-org-if/sources/my-source-id/documents
+     *                  </pre>
+     *
+     * @throws MalformedURLException
+     */
+    public PushSource(String apiKey, URL sourceUrl) throws MalformedURLException {
+        ApiUrl parser = new ApiUrl(sourceUrl);
+        PlatformUrl platformUrl = parser.getPlatformUrl();
+        String organizationId = parser.getOrganizationId();
+        this.sourceId = parser.getSourceId();
+        this.platformClient = new PlatformClient(apiKey, organizationId,
+                platformUrl);
+    }
+
+    /**
+     * Create a Push source instance from its
+     * <a href="https://docs.coveo.com/en/3295#stream-api-url">Stream API URL</a>
+     *
+     * @param apiKey         The API key used for all operations regarding your
+     *                       source.
+     *                       <p>
+     *                       Ensure your API key has the required privileges for the
+     *                       operation you will be performing
+     *                       *
+     *                       <p>
+     *                       For more information about which privileges are
+     *                       required,
+     *                       see
+     *                       <a href=
+     *                       "https://docs.coveo.com/en/1707#sources-domain">Privilege
+     *                       Reference.</a>
+     *
+     * @param organizationId The unique identifier of your organization.
+     *                       <p>
+     *                       The Organization Id can be retrieved in the URL of your
+     *                       Coveo organization.
+     *
+     * @param sourceId       The unique identifier of the target Push source.
+     *                       <p>
+     *                       The Source Id can be retrieved when you edit your
+     *                       source in the <a href=
+     *                       "https://docs.coveo.com/en/183/glossary/coveo-administration-console">Coveo
+     *                       Administration Console</a>
+     *
+     */
+    public PushSource(String apiKey, String organizationId, String sourceId) {
+        PlatformUrl platformUrl = new PlatformUrl(PlatformUrl.DEFAULT_ENVIRONMENT, PlatformUrl.DEFAULT_REGION);
+        this.sourceId = sourceId;
+        this.platformClient = new PlatformClient(apiKey, organizationId,
+                platformUrl);
+    }
```

Comment:
Almost all of the parameters in all these overloads are strings, I'm worried that adding new ways to create a push source could eventually be limited by the fact that other overloads already exist with the same new signature.

Maybe it would be a good idea to make the constructor private use named static methods to instantiate this? Like `PushSource.fromSourceUrl(URL sourceUrl, String apiKey)`, `PushSource.fromDefaultPlatformUrl(String organizationId, String sourceId, String apiKey)`, and `PushSource.fromPlatformUrl(String organizationId, String sourceId, String apiKey, PlatformUrl platformUrl)`.

Alternatively, a builder would be similarly descriptive, but I think it would be overkill and doesn't error at compile-time AFAIK.